### PR TITLE
KFSPTS-28175: Convert AccountReversion, AccountReversionDetail, and ReversionCategory to new lookup-inquiry framework and using new framework collections. Required creation of new docType ARVC.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

**Comment by nkimble** [comment]
Related SQL PR #1232 must be merged at the same time

**Comment by nkimble** [comment]
retest this please

File: src/main/resources/edu/cornell/kfs/coa/document/datadictionary/ReversionCategoryMaintenanceDocument.xml [hunk 1]
```diff
@@ -1,62 +1,47 @@
...
+    <bean id="ReversionCategoryMaintenanceDocument-EditReversionCategory" parent="ReversionCategoryMaintenanceDocument-EditReversionCategory-parentBean"/>
+    <bean id="ReversionCategoryMaintenanceDocument-EditReversionCategory-parentBean" abstract="true" parent="MaintainableSectionDefinition"
+          p:id="Edit Reversion Category"
+          p:title="Edit Reversion Category">
+        <property name="maintainableItems">
+            <list>
+                <bean parent="MaintainableFieldDefinition" p:name="reversionCategoryCode" p:required="true" p:newLookup="true"/>
+                <bean parent="MaintainableFieldDefinition" p:name="reversionCategoryName" p:required="true" p:newLookup="true"/>
+                <bean parent="MaintainableFieldDefinition" p:name="reversionSortCode" p:required="true" p:newLookup="true"/>
+                <bean parent="MaintainableFieldDefinition" p:name="active" p:defaultValue="true" p:required="true" p:newLookup="true"/>
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/coa/document/datadictionary/ReversionCategoryMaintenanceDocument.xml:38]: Can these fields omit the `newLookup` setting? It doesn't look like these four fields are linking to any other BOs.

**Review by cah292 — CHANGES_REQUESTED** [review]
Changes look good overall, though I did find some areas that could use further revisions.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml [hunk 2]
```diff
@@ -102,6 +145,72 @@
     </bean>
 
 
+    <!-- Field Definitions -->
+
+    <bean id="AccountReversion-universityFiscalYear-fieldDefinition"
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml]: I have revert to using the inline FieldDefinition notation per your request. 
I was following the example of how KualiCo converted collections to the new framework using that -fieldDefiniton setup as they never provide explanations of why they do what they do. We will need to keep in mind that we may have to revert back to that notation should they start to depend upon it in base code for some reason.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversionDetail.xml [hunk 3]
```diff
@@ -1,105 +1,130 @@
...
+            <ref bean="HiddenControl" />
+        </property>
+    </bean>
+
+    <bean id="AccountReversionDetail-newCollectionRecord" parent="GenericAttributes-newCollectionRecord"/>
+
+
+    <!-- Field Definitions -->
+
+    <bean id="AccountReversionDetail-accountReversionCategoryCode-fieldDefinition"
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversionDetail.xml]: Adjusted the code per your request.

**Comment by nkimble** [comment]
Requested changes have been made. Unit tests are running. This is ready to be reviewed again. Please also re-review the associated SQL for this code PR as I fixed a SQL issue identified during that SQL's PR review.

**Review by cah292 — COMMENTED** [review]
I noticed a couple more items that may need to be cleaned up, but overall the latest revisions look good.

File: src/main/java/edu/cornell/kfs/coa/businessobject/AccountReversion.java [hunk 4]
```diff
@@ -132,20 +83,17 @@ public void setAccount(Account account) {
...
-    }
-
-
     @Override
     public String getSourceAttribute() {
         
         return accountNumber;
     }
+
+    public String getAccountReversionViewer() {
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/coa/businessobject/AccountReversion.java:92]: Changed back.

File: src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml [hunk 5]
```diff
@@ -6,31 +6,55 @@
 
     <import resource="classpath:org/kuali/kfs/sys/sys-lookup-beans.xml"/>
     <import resource="classpath:edu/cornell/kfs/sys/cu-sys-attribute-beans.xml"/>
-    
+    <import resource="classpath:edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversionDetail.xml"/>
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/coa/businessobject/datadictionary/AccountReversion.xml:9]: removed import.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, my concerns have been addressed and the related SQL PR is also ready. (We decided that these changes can move ahead in advance of the documentation revisions.) If we discover that further cleanup is needed, we can handle that in a subsequent ticket.

Merging.

**Pull request merged**